src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidator.java

```diff
@@ -1,60 +1,117 @@
+        if (last >= 0) {
+            resultMessage = output.substring(last, output.length());
+            if (resultMessage.contains(ERROR_RESULT_BEGINNING)) {
+                validationError = resultMessage.replace(ERROR_RESULT_BEGINNING, "").trim();
+            }
         }
 
-        return true;
+        return buildSingleValidationResult();
     }
 
-    private File getFileToValidate() {
-        if (fileToValidate == null) {
-            if (filePath == null) {
-                throw new IllegalStateException(ErrorMessages.FILE_PATH_IS_EMPTY_OR_NULL);
+    String executeValidationAndGetResult() throws IOException, InterruptedException {
+        String commandToExecuteValidation = assembleValidatorCommand();
+
+        StringBuilder output = new StringBuilder();
+
+        Runtime rt = Runtime.getRuntime();
+        Process process = rt.exec(commandToExecuteValidation);
+        process.waitFor();
+        BufferedReader reader =
+                new BufferedReader(new InputStreamReader(process.getInputStream()));
+
+        String line;
+        while ((line = reader.readLine())!= null) {
+            if (output.length() > 0) {
+                output.append("\n");
             }
-            this.fileToValidate = findFileByTargetPath();
+            output.append(line);
         }
 
-        return fileToValidate;
+        return output.toString();
     }
 
-    private File findFileByTargetPath() {
-        return fileRepository.findByTargetPath(filePath);
+    private String assembleValidatorCommand() {
+        return String.join(" ", validatorPath, getCommandLineParams().getFilePath());
     }
 
-    public boolean isInValidStatusForContentValidation() {
-        if (!validStatusesForContentValidation().contains(getFileToValidate().getStatus())) {
-            throw new IllegalStateException(
-                    String.format(ErrorMessages.FILE_IN_ILLEGAL_STATE_MESSAGE, fileToValidate.getFilename()));
-        }
+    boolean validateParameters() {
+        validateFileExistence();
+
+        validateFileType();
 
         return true;
     }
 
-    private List<FileStatus> validStatusesForContentValidation() {
-        return Arrays.asList(FileStatus.READY_FOR_CHECKSUM, FileStatus.READY_FOR_ARCHIVE);
+    private void validateFileExistence() {
+        File file = new File(getCommandLineParams().getFilePath());
+        if(!file.exists() || file.isDirectory()) {
+            throw new FileNotFoundException(getCommandLineParams().getFilePath());
+        }
+    }
+
+    private void validateFileType() {
+        if (!FileType.forName(getCommandLineParams().getFileType())) {
+            throw new NotSupportedFileTypeException(getCommandLineParams().getFileType());
+        }
+    }
+
+    SingleValidationResult buildSingleValidationResult() {
+        SingleValidationResult singleValidationResult =
+                new SingleValidationResult(ValidationAuthor.FileContent, UUID.randomUUID().toString());
```

Comment:
Could you add a comment to explain why there's a random UUID here?

Reply:
I need to fix this and replace it with the file's UUID